sqlglot/generator.py

```diff
@@ -1475,7 +1476,15 @@ def in_unnest_op(self, unnest: exp.Unnest) -> str:
 
     def interval_sql(self, expression: exp.Interval) -> str:
         this = self.sql(expression, "this")
-        this = f" {this}" if this else ""
+        if this:
```

Comment:
this section doesn't look correct, you shouldn't call parse_one

Reply:
this = f" {this}" if isinstance(this, exp.Literal) or isinstance(this, exp.Paren) else f" ({this})"